### DISCUSSION TITLE
Trying to run the tutorial Hovering (hello, world)

### DISCUSSION THREAD

**Discussion by abdullja** [USC-ACTLab/crazyswarm#525] [Q&A]
I'm using Ros noetic, ubuntu 20, and using vrpn to track the crazyflie. I currently only have 1 crazyflie

The test script on simulation works. My issue is with the second part. When I run mocap_helper.launch I get the correct information and everything seems to be working. 

1. Rviz doesn't show the position of the crazyflie
2. I'm getting the following error 
![2021-11-07_23-02](https://user-images.githubusercontent.com/79109029/140660111-f2126151-2640-42f2-953b-643de43f6006.png)

This is what my hover_swarm.launch file looks like:
![2021-11-07_23-04](https://user-images.githubusercontent.com/79109029/140660154-b9c2d7f4-062f-44d0-af0b-765277d668c1.png)

**Comment by abdullja** [comment]
I figured that out now I'm getting this error and the crazyflie takes off but flips 
![2021-11-08_09-04](https://user-images.githubusercontent.com/79109029/140692062-f218a3c1-7edc-48d4-ad53-1dd30004f764.png)

**Reply by whoenig** [reply]
What is your output of `mocap_helper.launch`? This warning usually means that you didn't name your drone in your tracking software correctly.

**Reply by abdullja** [reply]
when I run mocap helper I get the correct position and orientation under the name cf1. I'm using an older version of motive so I created a ridged body and used vrpn. mocap_helper.launch seems to be running correctly.

**Reply by whoenig** [reply]
Please post the output of `mocap_helper.launch`. Also note that there is now support for older Motive versions via optitrack_closed_source, see the documentation for details.

**Reply by abdullja** [reply]
Thanks for the prompt response! Sure here's my output for mocap_helper.launch using vrpn. 
![2021-11-08_09-04](https://user-images.githubusercontent.com/79109029/140897873-f237708b-3249-48d9-8293-282104a7d351.png)

When I switch to optitrack_closed_source I get the following error:
![2021-11-09_12-24](https://user-images.githubusercontent.com/79109029/140897889-4b93ffcf-1110-4c6d-8d08-2b4d5dcb4626.png)

**Reply by whoenig** [reply]
For VRPN the output looks good, as you stated. Assuming that you use the same settings in the hover_swarm.launch file, I'd need to test that myself.
For optitrack_closed_source, this error indicates "network issues" according to the vendor. Perhaps you can `./sampleClient` from https://github.com/whoenig/NatNetSDKCrossplatform and see if that works for you or provides more details?

**Reply by abdullja** [reply]
Thank you for helping me through this. I'm pretty new to ros where am i suppose to download this file? inside ros_ws/src? Its giving me this when I try to make.
![2021-11-08_09-04](https://user-images.githubusercontent.com/79109029/140918396-678c0fe1-6c8a-4a66-8618-ea32cd1583fb.png)

**Reply by whoenig** [reply]
No, you should put it separately, (e.g., your home folder). `git clone` the repo and follow the instructions at https://github.com/whoenig/NatNetSDKCrossplatform#build.

**Reply by abdullja** [reply]
I get this error when I run ./sampleClient, and the same errors for both vrpn and optitrack_closed_source. 
![2021-11-08_09-04](https://user-images.githubusercontent.com/79109029/140936890-bfbce80b-fd9b-441b-b7b1-e3414e412cb4.png)

I know my there's nothing wrong with the hardware because I can fly using cfclient without any issues.

**Reply by whoenig** [reply]
cfclient doesn't use the motion capture system... You may contact OptiTrack directly with your problem. To me, it sounds like your network is not properly configured (i.e., your Ubuntu system is not on the same subnet as the Motive computer).

**Reply by abdullja** [reply]
That's weird because our OptiTrack has been working for all our other projects. I'll keep looking into this, but thank you for all your help!

**Reply by abdullja** [reply]
So here are my new findings:
1) I'm using vrpn and ros so I should be getting anything from NatNet anyway right? My ubuntu is connected to the same network as the computer running motive so there shouldn't be an issue there.
2) My motive is publishing at 120 hz but when I checked the vrpn.h file it says its publishing at 100 Hz is that ok?
3) I realized that when I run mocap.helper.launch I'm missing data for ever other frame. (Refer to the picture above) Is that normal?

What are your thoughts on all of this?

**Reply by abdullja** [reply]
I also get this error when i try to run the older version crazyflie_ros
![41659453-426165e8-7491-11e8-8bd0-9f70adabab5f](https://user-images.githubusercontent.com/79109029/141125997-ed5d2fa0-e5be-4766-aebc-ba272689c9cb.png)

**Reply by whoenig** [reply]
* Crazyflie_ros is not supported anymore.
* VRPN is not well tested I think, because I always use one of the other, more specialized backends. That being said, what you describe does sound like a bug. You should not be missing any data and ideally the framerates should match.

**Reply by abdullja** [reply]
I'm trying to recreate the environment you have in order to get this set up. Have you tested this with optitrack? what version of motive are you running?

**Reply by whoenig** [reply]
I am currently using Motive 3.0 (beta), but this also works with the latest stable (2.3.).

**Reply by wangmenghetrue** [reply]
Hi！Have you solved your problem? ‘No updated pose for Motion Capture Object CF1 ‘. How did you solve the problem?

**Reply by hit-JB** [reply]
I have the same problem as you before When I use the vicon motioncap just change the crazyserver params libobjectTrack to motioncap the issue solved

**Comment by powehi2020** [comment]
so? the problem dose not solved?
